pip_api/_parse_requirements.py

```diff
@@ -204,7 +204,8 @@ def parse_requirements(filename, options=None, include_invalid=False):
 
             elif known.requirements:
                 if known.requirements not in parsed:
-                    to_parse.add(known.requirements)
+                    file_location = os.path.dirname(filename)
```

Comment:
We only need to set `file_location` once, at the beginning of this method, not every single time this loop iterates.

Also `file_directory`, `path`,  `relative_path` or just `dirname` might be better names for this value.

Reply:
It seems that it should be set once for each file as it gets parsed, so I moved it to the top of the while loop.

And naming things is hard.  :P